target/linux/kirkwood/base-files/lib/upgrade/platform.sh

```diff
@@ -16,6 +27,16 @@ platform_do_upgrade() {
 	linksys,ea4500)
 		platform_do_upgrade_linksys "$1"
 		;;
+	netgear,readynas-duo-v2)
+		local root_mtd=$(cat /tmp/mounts | grep "/dev/root")
+		if [ -z "$root_mtd" ]; then
+			mkdir -p /var/lock
+			touch /var/lock/fw_printenv.lock
+			fw_setenv bootargs 'console=ttyS0,115200n8 earlyprintk'
+			fw_setenv bootcmd 'nand read.e 0x1200000 0x200000 0x600000;bootm 0x1200000'
```

Comment:
It's still lingering in my mind. I think this can be done manually once during the installation. We should access u-boot shell for installation anyway, why should we split the process?
I mean, `setenv bootcmd ... && saveenv` could be inserted somewhere around `bootm 0x1200000` in the commit message.

Reply:
Ok. I removed it.

Reply:
And then we don't need `platform_check_image()` change anymore, right?

Reply:
I was too fast. Now should be ok.